scrapy_zyte_api/providers.py

```diff
@@ -112,10 +124,25 @@ async def __call__(  # noqa: C901
                     options["extractFrom"] = extract_from.value
                     break
 
+        extract_from = None  # type: ignore[assignment]
         for item_type, kw in item_keywords.items():
             options_name = f"{kw}Options"
             if item_type not in to_provide_stripped and options_name in zyte_api_meta:
                 del zyte_api_meta[options_name]
+            elif options_name in zyte_api_meta:
+                extract_from = zyte_api_meta[options_name]["extractFrom"]
+
+        if AnyResponse in to_provide:
+            if extract_from == "browserHtml":
+                html_requested = True
```

Comment:
Does the new logic still need this flag?

Reply:
Currently, it's still being used. Though it might not be since I'm planning to simplify and refactor this already massive `__call__()` method we have.